src/vs/platform/backup/electron-main/backupMainService.ts

```diff
@@ -21,13 +23,29 @@ export class BackupMainService implements IBackupMainService {
 	private workspacesJsonContent: IBackupWorkspacesFormat;
 
 	constructor(
-		@IEnvironmentService environmentService: IEnvironmentService
+		@IEnvironmentService environmentService: IEnvironmentService,
+		@ILifecycleService lifecycleService: ILifecycleService
 	) {
 		this.backupHome = environmentService.backupHome;
 		this.workspacesJsonPath = environmentService.backupWorkspacesPath;
+
+		if (lifecycleService) {
```

Comment:
@Tyriar I am not a big fan of this pattern where we check for services to be present. if you have a test, make sure it includes a sufficient test lifecycle service.